File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> countMap = new HashMap<>();

        for (int n : nums) {
            countMap.put(n, countMap.getOrDefault(n, 0) + 1);
        }

        PriorityQueue<Map.Entry<Integer, Integer>> minHeap = new PriorityQueue<>(
            (e1, e2) -> e1.getValue() - e2.getValue());

        for (Map.Entry<Integer, Integer> entry : countMap.entrySet()) {
            minHeap.offer(entry);
            if (minHeap.size() > k) {
                minHeap.poll();
            }
        }

        int[] result = new int[k];

        for (int i=0; i<k; i++) {
            result[i] = minHeap.poll().getKey();
        }

        return result;
    }
}

File: Data Structures & Algorithms/single-number/submission-0.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        i = 0
        while i < len(nums):
            j = i + 1
            while j < len(nums):
                if nums[i] == nums[j]:
                    nums[i], nums[j] = 2147483647, 2147483647
                    break
                else:
                    j += 1
            i += 1

        res = 0
        for n in nums:
            if n != 2147483647:
                res = n

        return res
            